web-server/pages/api/resources/orgs/[org_id]/teams/v2.ts

```diff
@@ -302,3 +308,48 @@ const updateReposWorkflows = async (
     }
   }
 };
+
+const getTeamReposMap = async (org_id: ID, providers: Integration[]) => {
+  const baseQuery = db(Table.OrgRepo)
+    .where('org_id', org_id)
+    .andWhere('OrgRepo.is_active', true)
+    .whereIn('OrgRepo.provider', providers);
+
+  const teamJoin = baseQuery
+    .leftJoin({ tr: Table.TeamRepos }, 'OrgRepo.id', 'tr.org_repo_id')
+    .andWhere('tr.is_active', true);
+
+  const workflowJoin = teamJoin.leftJoin(
+    { rw: Table.RepoWorkflow },
+    function () {
+      this.on('OrgRepo.id', 'rw.org_repo_id').andOn(
+        'rw.is_active',
+        'tr.is_active'
+      );
+    }
+  );
```

Comment:
@Kamlesh72 Although this is perfectly fine, could you refactor it so that the left join doesn't use a function as the second parameter? This is because we've used left joins in our codebase, but haven't used this type of functional mapping before.

This is purely an optional request.

Reply:
@Kamlesh72 everything else looks good

Reply:
@shivam-bit will this be ok?

```suggestion
const workflowJoin = teamJoin
  .leftJoin({ rw: Table.RepoWorkflow }, 'OrgRepo.id', 'rw.org_repo_id')
  .andWhere(function () {
    this.where('rw.is_active', true).orWhereNull('rw.is_active');
  });
```

Reply:
Also can you confirm if we need to mark corresponding TeamRepo and OrgRepo is_active to false if a Team is deleted. The OrgRepo will be marked inactive if there is no active TeamRepo.

This can be either handled in nextjs server or there already exists a route to Delete a team in flask server (code needs to be updated).

Reply:
> Also can you confirm if we need to mark corresponding TeamRepo and OrgRepo is_active to false if a Team is deleted. The OrgRepo will be marked inactive if there is no active TeamRepo.
Yes we expect this behaviour, as it will prevent the sync server from syncing any repos which are not being used by any team

Reply:
> @shivam-bit will this be ok?

still same issue, but lets ignore that i checked the code base again seems like there are few files where we are using function as the second parameter in joins. so its cool.